internal/appsec/dyngo/instrumentation/common.go

```diff
@@ -64,3 +67,65 @@ func SetTags(span ddtrace.Span, tags map[string]interface{}) {
 		span.SetTag(k, v)
 	}
 }
+
+// SetEventSpanTags sets the security event span tags into the service entry span.
+func SetEventSpanTags(span ddtrace.Span, events []json.RawMessage) error {
+	// Set the appsec event span tag
+	val, err := makeEventTagValue(events)
+	if err != nil {
+		return err
+	}
+	span.SetTag("_dd.appsec.json", string(val))
+	// Keep this span due to the security event
+	//
+	// This is a workaround to tell the tracer that the trace was kept by AppSec.
+	// Passing any other value than `appsec.SamplerAppSec` has no effect.
+	// Customers should use `span.SetTag(ext.ManualKeep, true)` pattern
+	// to keep the trace, manually.
+	span.SetTag(ext.ManualKeep, samplernames.AppSec)
```

Comment:
Since we set the priority from the WAF too now with the first rulesetinfo metrics, can you please reuse this into a new shared function in this package that the WAF should be also able to use for now too?

Reply:
Could you elaborate?
The code in appsec/waf.go sets the priority through `op.AddTag` since it doesn't have access to the span, while this directly uses the span, so it seems like 2 different use cases to me. Am I missing your point?

Reply:
After discussion: using s`amplernames.AppSec` in _appsec/waf.go_ as well, and leaving the code untouched otherwise.